ITE/11/README.adoc

```diff
@@ -0,0 +1,278 @@
+= ITE-11: Verifying Attributes in in-toto Attestations
+:source-highlighter: pygments
+:toc: preamble
+:toclevels: 2
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+.Metadata
+[cols="2"]
+|===
+| ITE
+| 11
+
+| Title
+| Verifying Attributes in in-toto Attestations
+
+| Sponsor
+| link:https://github.com/adityasaky[Aditya Sirish A Yelgundhalli]
+
+| Status
+| Draft
+
+| Type
+| Standards
+
+| Created
+| 2023-01-07
+
+|===
+
+[[abstract]]
+== Abstract
+
+in-toto metadata, whether legacy links or contextual predicates defined under
+the Attestation Framework, can record various attributes about software supply
+chain steps and artifacts. However, layouts do not intrinsically support
+setting constraints on these attributes, passing off these capabilities to the
+"inspections" semantic instead. This ITE proposes adding support for setting
+such constraints in in-toto layouts.
+
+[[specification]]
+== Specification
```

Comment:
Sorry if I'm missing context, but how do you do graph traversal? For example, if you want to have a policy that says that the artifact must have a SLSA provenance attestation where each `resolvedDependency` has its own VSA attestation having some other properties?

Is that a general feature of in-toto layouts that is orthogonal to this?

Reply:
Somewhat, though it's not currently part of the spec. @marcelamelara has worked on this recently in the context of SCAI evidence references, to go resolve the referenced attestation for use in the verification process. I view this as possibly a preprocessor / ahead of time run, so at layout verification the required attestations are present. This may, however, become problematic with lots of references, only some of which are actually required. I'm open to fleshing this out some more, but I suspect this is a more general in-toto concern and not just for ITE-11.

Did I parse your question correctly?

Reply:
Yes, you parsed it correctly. So in short, that's a known limitation and this is a first step where you can make decisions that don't require graph traversal? And that a future ITE could bring better graph traversal options?